File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/CatalogEntry.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class CatalogEntryRelatedEntry(BackboneElement):
    item: Reference
    relationtype: str

@dataclass(kw_only=True)
class CatalogEntry(DomainResource):
    additional_characteristic: Optional[List[CodeableConcept]] = field(default_factory=list)
    additional_classification: Optional[List[CodeableConcept]] = field(default_factory=list)
    referenced_item: Reference
    type: Optional[CodeableConcept] = None
    classification: Optional[List[CodeableConcept]] = field(default_factory=list)
    validity_period: Optional[Period] = None
    orderable: bool
    status: Optional[str] = None
    valid_to: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    additional_identifier: Optional[List[Identifier]] = field(default_factory=list)
    last_updated: Optional[str] = None
    related_entry: Optional[List[CatalogEntryRelatedEntry]] = field(default_factory=list)

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/CatalogEntry.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class CatalogEntry : DomainResource, IResource
{
    public Base.CodeableConcept[]? AdditionalCharacteristic { get; set; }
    public Base.CodeableConcept[]? AdditionalClassification { get; set; }
    public required Base.ResourceReference ReferencedItem { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public Base.CodeableConcept[]? Classification { get; set; }
    public Base.Period? ValidityPeriod { get; set; }
    public required bool Orderable { get; set; }
    public string? Status { get; set; }
    public string? ValidTo { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.Identifier[]? AdditionalIdentifier { get; set; }
    public string? LastUpdated { get; set; }
    public CatalogEntry_RelatedEntry[]? RelatedEntry { get; set; }

    public class CatalogEntry_RelatedEntry : BackboneElement
    {
        public required Base.ResourceReference Item { get; set; }
        public required string Relationtype { get; set; }
    }
}